src/ports/postgres/modules/convex/utils_regularization.py_in

```diff
@@ -47,14 +46,22 @@ def __utils_dep_var_scale(**kwargs):
     The output will be stored in a temp table: a mean array and a std array
 
     This function is also used in lasso.
+
+    Parameters:
+    schema_madlib -- madlib schema
+    tbl_data -- original data
+    col_ind_var -- independent variables column
+    col_dep_var -- dependent variable column
     """
+
     y_scale = plpy.execute(
         """
         select
-            avg({col_dep_var}) as mean,
+            avg(case when not {schema_madlib}.array_contains_null({col_ind_var}) then {col_dep_var} end) as mean,
             1 as std
```

Comment:
@decibel that is a good point. I agree that for `avg()` function `col_dep_var` containing NULLs is not a problem. But what I meant is that it will be a problem for normalizing the `col_ind_var` because in that case we cannot treat `col_ind_var` and `col_dep_var` independently as we are currently doing.